### PR TITLE
Programming/movement

### PR DESCRIPTION
**Pull request by RSGameDev**
Here's what i've done up to this evening. 
I tried amending the scripts i worked on so it adopts the preferred code style for this project.

I still would like to make a little bit more alteration to the movement code but I wanted to push an update so you have what  has been worked on currently.

The movement has been focused on. Now there is lerping, a couple of things that worked ok before aren't working as well now. This is mainly player-enemy interaction for example. Sometimes when the player goes into the enemy it stays on the same tile as the enemy, instead of being pushed back. This is to do with the new lerp feature thats in the game.
So that's something we'll have to address at some point.

### CODE REVIEW

File: bpdv3/Assets/Scripts/Floor/TileController.cs [hunk 1]
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
 
     public GameObject[] tilesArray;
     public GameObject player;
-    
 
+    public bool tilesScrolling;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Floor/TileController.cs]: IsPlatformMoving is a better name and make it private if it doesn't have to be public, in that case _isPlatformMoving

File: bpdv3/Assets/Scripts/Floor/TileController.cs [hunk 2]
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
...
+    
     Vector3 direction;
     public float distance;
 
+    Vector3 tileMovement;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Floor/TileController.cs]: Please specify accessor and rename to _tileMovement

File: bpdv3/Assets/Scripts/Floor/TileController.cs [hunk 3]
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
...
+
     // Use this for initialization
     void Start()
     {
         direction = Vector3.back;
-        distance = direction.magnitude;
+        distance = 1.25f;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Floor/TileController.cs]: Please make 1.25f a constant

File: bpdv3/Assets/Scripts/Floor/TileController.cs [hunk 4]
```diff
@@ -15,26 +15,31 @@ public class TileController : MonoBehaviour {
...
     void Update()
     {
         if (SceneController.Instance.scrollBeatCount >= 6)
         {
+            tilesScrolling = true;
+
             foreach (GameObject gameObject in tilesArray)
             {
-                var tileMovement = direction.normalized * (Time.deltaTime * (1.25f / 4f));
+                tileMovement = direction * (Time.deltaTime * (distance / 3f));
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Floor/TileController.cs]: Please make 3f a constant

File: bpdv3/Assets/Scripts/Lose Condition/BottomRowHazard.cs [hunk 5]
```diff
@@ -15,7 +15,7 @@ private void OnTriggerStay(Collider other)
         if (other.gameObject.CompareTag("Player"))
         {
             other.GetComponent<Player>().PlayerDied();
-            other.GetComponent<Player>().StartPosition();
+            //other.GetComponent<Player>().StartPosition();
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Lose Condition/BottomRowHazard.cs]: Delete if unnecessary

File: bpdv3/Assets/Scripts/Mechanics/Anchor.cs [hunk 6]
```diff
@@ -7,21 +7,28 @@ public class Anchor : MonoBehaviour
 {
     public GameObject tileObj;
 
+    Vector3 newPos;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Mechanics/Anchor.cs]: please specify accessor and rename to _newPosition

File: bpdv3/Assets/Scripts/Mechanics/Anchor.cs [hunk 7]
```diff
@@ -7,21 +7,28 @@ public class Anchor : MonoBehaviour
...
+
+    [SerializeField] Vector3 tilepos;
+
     private void OnTriggerEnter(Collider other)
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Mechanics/Anchor.cs:20]: This method does too much job inside please seperate its responsilibities into smaller methods and make easy to read. It is impossible to understand what happens onTrigger

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 8]
```diff
@@ -6,24 +6,37 @@
 // Script attached to the player.
 public class Player : MonoBehaviour
 {
+    #region Private variables
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs:10]: Great :)

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 9]
```diff
@@ -6,24 +6,37 @@
...
+    public GameObject[] spawnPlayerDetect;
+
     public GameUI gameUIScript;
+    public Vector3 startPos;
 
     public int livesCountPlayer = 3;
 
-    bool isAlive;
-
+    public bool isAlive = true;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs]: PascalCase -> IsAlive

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 10]
```diff
@@ -6,24 +6,37 @@
...
 {
+    #region Private variables
+    private PlayerMovement _playerMovement;
+    #endregion
+
+    #region Public variables
+    public GameObject[] spawnPlayerDetect;
+
     public GameUI gameUIScript;
+    public Vector3 startPos;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs]: StartPosition

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 11]
```diff
@@ -6,24 +6,37 @@
 // Script attached to the player.
 public class Player : MonoBehaviour
 {
+    #region Private variables
+    private PlayerMovement _playerMovement;
+    #endregion
+
+    #region Public variables
+    public GameObject[] spawnPlayerDetect;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs]: Please find a more definitive name

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 12]
```diff
@@ -6,24 +6,37 @@
...
+
     public GameUI gameUIScript;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs]: Please use PascalCase on public variables

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 13]
```diff
@@ -6,24 +6,37 @@
...
+    public bool isAlive = true;
     public bool isPushBack;
-    
-    public Vector3 startPos;
+    #endregion    
 
-    public GameObject[] spawnPlayerDetect;
+    private void Awake()
+    {
+        _playerMovement = GetComponent<PlayerMovement>();
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs:27]: Add requiredComponent at the top of class

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 14]
```diff
@@ -35,12 +48,23 @@ public void StartPosition()
 
     public void PlayerDied()
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs]: KillPlayer()

File: bpdv3/Assets/Scripts/Player/Player.cs [hunk 15]
```diff
@@ -35,12 +48,23 @@ public void StartPosition()
...
         foreach (GameObject go in spawnPlayerDetect)
         {
             go.GetComponentInChildren<SpawnDetectPlayer>().playerInFront = false;
         }
+        StartPosition();
+        _playerMovement.ResetPosition();   // This is use so the lerp function does not continue to use the old values still.
+        StartCoroutine(EnableMove());
+    }
+
+    IEnumerator EnableMove()
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/Player.cs]: please specifty accessor

File: bpdv3/Assets/Scripts/Player/PlayerMovement.cs [hunk 16]
```diff
@@ -5,86 +5,136 @@
...
+        float fractionOfJourney = distCovered / journeyLength;
+        transform.position = Vector3.Lerp(transform.position, destination, fractionOfJourney);
+        if (fractionOfJourney >= 0.1f || !_player.isAlive)
+        {
+            lerp = false;
+            _anchor.PlaceInPosition();
+        }
+    }
 
+    void Movement()
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/PlayerMovement.cs]: please specift accessor

File: bpdv3/Assets/Scripts/Player/PlayerMovement.cs [hunk 17]
```diff
@@ -5,86 +5,136 @@
...
             transform.LookAt(transform.position + Vector3.right);
-            transform.position += new Vector3(1.25f, 0, 0);
-            if (transform.position.x > 5)
+            destination = new Vector3(transform.position.x + 1.25f, transform.position.y, transform.position.z);
+            journeyLength = Vector3.Distance(transform.position, destination);
+            lerp = true;
+        }
+    }
+
+    void CheckBoundaries()
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/PlayerMovement.cs]: Accessor

Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/PlayerMovement.cs]: Please remove magic numbers in this method and add comments to this method

File: bpdv3/Assets/Scripts/enemy/Enemy.cs [hunk 18]
```diff
@@ -6,52 +6,57 @@
...
+    private Anchor _anchor;
+    private EnemyMovement _enemyMovement;
+    private EnemyNextMove _nextMove;
+    #endregion
 
-    public bool isPushBack;
+    #region Public variables
+    public bool isPushBack; 
+    public bool isAlive = true;
+    public bool isNew = true;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/Enemy.cs]: IsPushBack, IsAlive, IsNew

File: bpdv3/Assets/Scripts/enemy/Enemy.cs [hunk 19]
```diff
@@ -73,11 +78,11 @@ private void OnTriggerEnter(Collider other)
 
     private void EnemyDies()
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/Enemy.cs]: KillEnemy();

File: bpdv3/Assets/Scripts/enemy/EnemyMovement.cs [hunk 20]
```diff
@@ -6,31 +6,68 @@
...
+    private float journeyLength;
+    public float speed = 1.0F;
+    public bool lerp;
     #endregion
 
     private void Awake()
     {
-        enemyScript = GetComponent<Enemy>();
-        nextMoveScript = transform.GetChild(2).GetComponent<EnemyNextMove>();
+        _enemy = GetComponent<Enemy>();
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/EnemyMovement.cs:48]: Use requeiredocmponent

File: bpdv3/Assets/Scripts/enemy/EnemyMovement.cs [hunk 21]
```diff
@@ -6,31 +6,68 @@
...
+
+        GameObject temp = GameObject.FindGameObjectWithTag("TileController");
+        _tileController = temp.GetComponent<TileController>();
+    }
+
+    void Update()
+    {
+        if (lerp)
+        {
+            Lerp();
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/EnemyMovement.cs]: Make your code readable for anyone not for only coders. Assume that I dont know how to code. Lerp is a mystery for me, please use a more self explanatory method name

File: bpdv3/Assets/Scripts/enemy/EnemyMovement.cs [hunk 22]
```diff
@@ -6,31 +6,68 @@
...
+    private bool hasMoved;  
+    #endregion
 
+    #region Function
+    // Lerp    
+    [SerializeField] Vector3 destination;
+    private float startTime;
+    private float journeyLength;
+    public float speed = 1.0F;
+    public bool lerp;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/EnemyMovement.cs]: What is the usage of this boolean, name does not explain

File: bpdv3/Assets/Scripts/enemy/EnemyMovement.cs [hunk 23]
```diff
@@ -77,7 +114,7 @@ public void Movement()
             }
         }
 
-        if (enemyScript.currentEnemyType == Enemy.EnemyType.Rook)
+        if (_enemy.currentEnemyType == Enemy.EnemyType.Rook)
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/EnemyMovement.cs]: An enemy cant be both rook and a shroom therfore use switch or if else

**Review by KEREMADALI — CHANGES_REQUESTED** [review]
Since it is a new convention for you, it is normal to mistake for some of them, please dont be discoured by my comments. As a reminder;

public  -> PascalCase
private -> _camelCase
local variable -> camelCase
Always define accessors both on variables and methods

**Review by RSGameDev — COMMENTED** [review]
Hi, I'll push the changes shortly. Will mark for review again.

**Review by RSGameDev — COMMENTED** [review]
Hi, this is a recent push with code review changes based from the last review.
Let me know if it appears better. Thanks Kerem.

File: bpdv3/Assets/Scripts/Floor/TileController.cs [hunk 24]
```diff
@@ -5,53 +5,57 @@
...
-        direction = Vector3.back;
-        distance = direction.magnitude;
+        _direction = Vector3.back;
     }
 
     // Update is called once per frame
     void Update()
     {
-        if (SceneController.Instance.scrollBeatCount >= 6)
+        if (SceneController.Instance.scrollBeatCount >= s_beatToStartScroll)
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Floor/TileController.cs:33]: All of code in here seems like they control the same thing u can create a method for this and that method in Update

File: bpdv3/Assets/Scripts/Mechanics/Anchor.cs [hunk 25]
```diff
@@ -5,39 +5,61 @@
...
+        }
 
-        if (gameObject.transform.parent.CompareTag("Player") && other.gameObject.layer == LayerMask.NameToLayer("Floor"))
+        if (tag == s_Player && other.gameObject.layer == LayerMask.NameToLayer(s_FloorLayer))
         {
-            tileObj = other.gameObject;
-            Vector3 newPos = other.GetComponent<Renderer>().bounds.center;
-            transform.parent.position = new Vector3(newPos.x, newPos.y + 1f, newPos.z);
-            transform.parent.SetParent(other.transform);
+            print("fsavda");
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Mechanics/Anchor.cs:31]: Please remove

File: bpdv3/Assets/Scripts/Mechanics/Anchor.cs (hunk 7 above)
Comment (KEREMADALI) [reply bpdv3/Assets/Scripts/Mechanics/Anchor.cs:20]: Good updates, you can add comments like "resolved" on fixes

File: bpdv3/Assets/Scripts/Player/PlayerMovement.cs [hunk 26]
```diff
@@ -2,89 +2,150 @@
...
+    private void SmoothMovement()
     {
-        Vector3 position;
-        position = transform.position;
+        float distCovered = (Time.time - _startTime) * s_Speed;
+        float fractionOfJourney = distCovered / _journeyLength;
+        transform.position = Vector3.Lerp(transform.position, _destination, fractionOfJourney);
+        if (fractionOfJourney >= 0.1f || !Player.IsPlayerAlive)
+        {
+            IsLerpingStarted = false;
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/Player/PlayerMovement.cs:64]: If we removede lerp can this bool deleted?

File: bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs [hunk 27]
```diff
@@ -5,31 +5,30 @@
...
 
     public int CheckNumber()
     {
-        occupiedNum = tileObj.GetComponent<TileProperties>().occupiedNum;        
-        return occupiedNum;
+        OccupiedNumber = EnemyNextMoveTileObject.GetComponent<TileProperties>().occupiedNum;        
+        return OccupiedNumber;
     }
 
     public bool CanMove()
```
Comment (KEREMADALI) [inline bpdv3/Assets/Scripts/enemy/EnemyNextMove.cs:29]: CanEnemyMove is a better name for this method if it is being used by only the enemy

**Review by KEREMADALI — COMMENTED** [review]
Good updates well done

**Comment by RSGameDev** [comment]
Thanks for the further comments, ill look at them fully over next couple days and will amend the code accordingly from the notes u have left. I'll look to make the shroom movement better working also as at the moment. It's not in the best shape ^^. Is getting there :). Just having a resting day today, not feeling too well after last night lol.

**Comment by RSGameDev** [comment]
Corrected changes. i forgot in a recent push but have done it for the push im about to do. Thanks for the review also.

**Pull request closed without merging**